internal/controllers/addon/olm_reconciler.go

```diff
@@ -55,6 +55,14 @@ func (r *olmReconciler) Reconcile(ctx context.Context,
 	}
 
 	// Phase 4.
+	// Ensure NetworkPolicy for CatalogSources
+	if requeueResult, err = r.ensureCatalogSourcesNetworkPolicy(ctx, addon); err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to ensure NetworkPolicy for CatalogSources: %w", err)
+	} else if requeueResult != resultNil {
+		return handleExit(requeueResult), nil
+	}
```

Comment:
I think this should be a separate sub-reconciler itself (`networkPolicyReconciler`) since it is not really an OLM-related object

Reply:
I agree that `NetworkPolicies` in general are unrelated to OLM, but this particular `NetworkPolicy` is only required because of OLM's need to access `CatalogSources`. Maybe if we end up having more `NetworkPolicies` we would refactor this out, but then we have dependencies farther from where they are required. Kind of the same reason why a dedicated `Namespace` reconciler is weird since you`d end up with multiple reconcilers dependent on that one reconciler to do their work. I think something that would be nice is a common toolset to reconcile particular resources that each reconciler could make use of. That way there is a standard `ensureXresource` that each domain scoped reconciler can leverage.